File: p1/594/main.py
import sys

if __name__ == "__main__":
    numbers = [int(line.strip()) for line in sys.stdin.readlines()]

    for number in numbers:
        b1 = number        & 0xFF
        b2 = number >>   8 & 0xFF
        b3 = number >> 2*8 & 0xFF
        b4 = number >> 3*8 & 0xFF

        swapped = b1 << 3*8 | b2 << 2*8 | b3 << 8 | b4 

        if swapped & (0x1 << 31):
            swapped = -(1 + (swapped ^ 0xFFFFFFFF))
        print(number, "converts to", swapped)

File: p1/594/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);

        int x;
        while(s.hasNextInt()) {
            x = s.nextInt();

            int b1 = x & 0xFF;
            int b2 = (x >> 8) & 0xFF;
            int b3 = (x >> 16) & 0xFF;
            int b4 = (x >> 24) & 0xFF;
            
            int y = (b1 << 24) | (b2 << 16) | (b3 << 8) | b4;
            if ((y & (1 << 31)) == 1)
                y = -(1 + (y ^ 0xffffffff));

            System.out.format("%d converts to %d\n", x, y); 
        }

        s.close();
    }
}
